File: 53-MaximumSubarray/53-MaximumSubarray.java
// Last updated: 6/26/2026, 8:55:19 AM
class Solution {
    public int maxSubArray(int[] nums) {
       int max=nums[0];
       int sum=0;
        for(int i=0;i<nums.length;i++){
              sum+= nums[i];
              max=Math.max(sum,max);
            if(sum<0){
                sum=0;
            }
       }
        return max;
    }
}

File: 53-MaximumSubarray/53-MaximumSubarray.js
// Last updated: 6/26/2026, 8:55:15 AM
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let output = -Infinity;
        let tempMax = 0;
        for (let i =0; i< nums.length; i++) {
            tempMax = tempMax + nums[i];
            output = Math.max(output, tempMax);
            if(tempMax < 0) tempMax = 0;
        }
        return output;
};